### PR TITLE
Timeline history graph - react to click to display entity more info

### PR DESCRIPTION
**Pull request by k3a**
## Proposed change

The timeline history graph card is great, however it lacks details.
Sometimes the intervals between changes are short (as in door sensors), and sometimes the sensor is unavailable for a short period of time, which is not very obvious from the graph.

Therefore, I enhanced the chart by adding a small inline plugin to handle clicking on a timeline element and firing  "hass-more-info" to display the More info dialog of the displayed entity.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/chart/ha-chart-base.ts [hunk 1]
```diff
@@ -288,6 +329,25 @@ export class HaChartBase extends LitElement {
           display: false,
         },
       },
+      {
+        id: "labelClickHandler",
```
Comment (bramkragten) [inline src/components/chart/ha-chart-base.ts:333]: This should not be in `ha-chart-base`, use the `plugin` property to add a plugin to the chart from just the place you need it.

See for example:

https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/cards/energy/hui-energy-devices-graph-card.ts#L156-L166

**Comment by k3a** [comment]
Thanks for the suggestion! I have updated it to match the way `hui-energy-devices-graph-card` does it (clicking on a history bar element instead of a label). It is probably batter to have it consistent and I tested that it works fine on mobile too. Tapping opens More info, long tap keeps tooltip only.

It is now implemented in `state-history-chart-timeline` which should be the best spot.
The commit is much smaller too.

**Comment by bramkragten** [comment]
We should prevent the action when the graph is the more info itself, as it will give a weird behaviour of opening the more info again.

**Comment by k3a** [comment]
I have added `inMoreInfo` (default False) property to `state-history-chart-timeline` and `state-history-charts`.
It is quite specific property for "is in more info dialog" which I don't like that much but this is really the only place where this more-info event should not be fired.

I was considering adding `clickForMoreInfo` (default True) at first. That is more universal and nicer but having it as a property of `state-history-charts` would kind of imply that it applies for `state-history-chart-line` too.
But having more-info handler on `state-history-chart-line` doesn't seem useful to me because More info dialog won't provide more info than that line chart, except maybe the button to access more history.

If you think more-info handler could be useful for the line history graph too, I can rework it into `clickForMoreInfo` property.

File: src/dialogs/more-info/ha-more-info-history.ts [hunk 2]
```diff
@@ -99,6 +99,7 @@ export class MoreInfoHistory extends LitElement {
                 .historyData=${this._stateHistory}
                 .isLoadingData=${!this._stateHistory}
                 .showNames=${false}
+                .inMoreInfo=${true}
```
Comment (bramkragten) [inline src/dialogs/more-info/ha-more-info-history.ts:102]:
```suggestion
                inMoreInfo
```

**Review by bramkragten — APPROVED** [review]
Thanks!

**Comment by k3a** [comment]
Actually please take a look at alternative PR #18036 which supports both chart types. I think it makes more sense unless you plan to have some other clicking function on line history state chart in the future.

**Pull request closed without merging**